### PR TITLE
fix: Form.Item with `help` should display error style when validate failed

### LINKED ISSUE #25560: Form.Item does not display validation message when set help
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/hardcore-haze-kqo2l)

### Steps to reproduce
- Click Submit

### What is expected?
- Validation trigger

### What is actually happening?
- Validation not trigger

| Environment | Info |
|---|---|
| antd | 4.4.1 |
| React | The latest |
| System | mac |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by zhangchen915**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #25560 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Form.Item `help` style when verification fails  |
| 🇨🇳 Chinese |   修复 Form.Item  `help`  在校验失败时的样式    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by fireairforce — APPROVED** [review]
LGTM

**Review by xrkffgg — COMMENTED** [review]
PR里的 changelog 填写一下

File: components/form/FormItem.tsx [hunk 1]
```diff
@@ -159,15 +159,10 @@ function FormItem(props: FormItemProps): React.ReactElement {
...
-    } else if (meta && meta.validating) {
-      mergedValidateStatus = 'validating';
-    } else if (!help && mergedErrors.length) {
-      mergedValidateStatus = 'error';
-    } else if (meta && meta.touched) {
-      mergedValidateStatus = 'success';
-    }
+    if (validateStatus !== undefined) mergedValidateStatus = validateStatus;
+    else if (meta?.validating) mergedValidateStatus = 'validating';
+    else if (meta?.errors?.length) mergedValidateStatus = 'error';
```
Comment (xrkffgg) [inline components/form/FormItem.tsx:164]: mergedErrors ？

Comment (zhangchen915) [reply components/form/FormItem.tsx:164]: Form.Item with `help` 
`mergedErrors = toArray(help);`

Comment (zombieJ) [reply components/form/FormItem.tsx:164]: should keep `{ }`.

Comment (zhangchen915) [reply components/form/FormItem.tsx]: Done

**Comment by crazyair** [comment]
不知道是不是这个修改导致了现在的问题

image

https://codesandbox.io/s/strange-babbage-hkq1n?file=/index.js

**Comment by zombieJ** [comment]
嗯，修过了 #25737

**Pull request merged**